app/services/measurements_creator.rb

```diff
@@ -2,6 +2,13 @@ class MeasurementsCreator
   SLICE_SIZE = 500
 
   def self.call(stream, measurements_attributes)
+    if stream.session.type == "FixedSession"
+      measurements_attributes = measurements_attributes.map do |measurement_attributes|
+        measurement_attributes[:utc_time] = Time.now.utc
```

Comment:
Not sure in this case but may be better `Time.current` (https://robots.thoughtbot.com/its-about-time-zones)